### PR TITLE
Adding the Memory Maps chapter for `dm`, `dmi`, `dmm`, etc

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
Added a chapter for Memory Maps, resolving issues #7 and #103.

### CODE REVIEW

File: debugger/memory_maps.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    | dmm[?][j*]                       List modules (libraries, binaries loaded in memory)
+    | dmp[?] <address> <size> <perms>  Change page at <address> with <size>, protection <perms> (rwx)
+    | dms[?] <id> <mapaddr>            Take memory snapshot
+    | dms- <id> <mapaddr>              Restore memory snapshot
+    | dmS [addr|libname] [sectname]    List sections of target lib
+    | dmS* [addr|libname] [sectname]   List sections of target lib in radare commands
+
+In this chapter, we'll go over some of the most useful subcommands of `dm` using simple examples. For the following examples, we'll use a simple `helloworld` program for Linux but it'll be the same for every binary.
+
+First thing first - open a program in debugging mode:
```
Comment (radare) [inline debugger/memory_maps.md]: first thing(s) first

File: debugger/memory_maps.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+    $ r2 -d helloworld
```
Comment (radare) [inline debugger/memory_maps.md]: explain that r2 will try to find this program in current directory and then in $PATH, even if no ./ is passed. this is contradictory with UNIX systems, but makes the behaviour consistent for windows users

File: debugger/memory_maps.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    Process with PID 20304 started...
```
Comment (radare) [inline debugger/memory_maps.md]: i filled an issue to remove all this debug information https://github.com/radare/radare2/issues/10756

File: debugger/memory_maps.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+We can see that along with our `helloworld` binary itself, another library was loaded which is `ld-2.27.so`. We don't see `libc` yet and this is because radare2 breaks before `libc` is loaded to memory. Let's use `dcu` (**d**ebug **c**ontinue **u**ntil) to execute our program until the entry point of the program, which radare flags as `entry0`.
+
+    [0x7fa80a19dfb0]> dcu entry0
+    Continue until 0x55ca23a4a520 using 1 bpsize
+    hit breakpoint at: 55ca23a4a518
+    [0x55ca23a4a520]> dmm
+    0x55ca23a4a000 /tmp/helloworld
+    0x7fa809de1000 /usr/lib/libc-2.27.so
+    0x7fa80a19d000 /usr/lib/ld-2.27.so
```
Comment (radare) [inline debugger/memory_maps.md]: this output is different depending on the operating system

File: debugger/memory_maps.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    17 0x00025f78     0 0x7fae46752f80   376 -rw- ld-2.27.so..bss
+    18 0x00025f78    17 0x00000000    17 ---- ld-2.27.so..comment
+    19 0x00025fa0    63 0x00000000    63 ---- ld-2.27.so..gnu.warning.llseek
+    20 0x00025fe0 13272 0x00000000 13272 ---- ld-2.27.so..symtab
+    21 0x000293b8  7101 0x00000000  7101 ---- ld-2.27.so..strtab
+    22 0x0002af75   215 0x00000000   215 ---- ld-2.27.so..shstrtab
+
+radare2's `dm` subcommands can also display a map of the heap which is useful for those who are interesting in inspecting the heap and its content. Simply execute `dmh` to show a map of the heap:
+
+    [0x7fae46236ca6]> dmh
```
Comment (radare) [inline debugger/memory_maps.md]: probably put all the heap stuff in a separate chapter

**Comment by radare** [comment]
please explain the dm. and dmi. commands to get information of the current offset

**Comment by radare** [comment]
also the dm= is handy to see how those maps are located in memory

File: debugger/memory_maps.md (hunk 3 above)
Comment (ITAYC0HEN) [reply debugger/memory_maps.md]: I can remove from the example all the lines between "$ r2 -d helloworld" and the prompt. Is this the expected output of the fix to the issue? Don't you want to keep one of them (starting the process, attaching to the process)?

File: debugger/memory_maps.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    15 0x00024f98    96 0x7fae46751f98    96 -rw- ld-2.27.so..got
+    16 0x00025000  3960 0x7fae46752000  3960 -rw- ld-2.27.so..data
+    17 0x00025f78     0 0x7fae46752f80   376 -rw- ld-2.27.so..bss
+    18 0x00025f78    17 0x00000000    17 ---- ld-2.27.so..comment
+    19 0x00025fa0    63 0x00000000    63 ---- ld-2.27.so..gnu.warning.llseek
+    20 0x00025fe0 13272 0x00000000 13272 ---- ld-2.27.so..symtab
+    21 0x000293b8  7101 0x00000000  7101 ---- ld-2.27.so..strtab
+    22 0x0002af75   215 0x00000000   215 ---- ld-2.27.so..shstrtab
+
+## Memory map of the Heap
```
Comment (radare) [inline debugger/memory_maps.md]: put that in a different file, otherwise the index doesnt looks good. debugger/heap.md

Comment (ITAYC0HEN) [reply debugger/memory_maps.md]: fixed

**Pull request merged**